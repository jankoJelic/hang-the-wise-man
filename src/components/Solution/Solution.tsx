import onlyLettersInString from "util/onlyLettersInString";

type Props = {
  solution: string;
  usedLetters: string[];
};

const Solution = ({ solution, usedLetters }: Props) => {
  return (
    <div className="w-full flex flex-wrap items-center px-8">
      {solution.split("").map((char, index) => {
        const isLetter = onlyLettersInString(char);
        const isUsed = usedLetters.includes(char.toUpperCase());
        const isQuotationMark = char === "'";
        const isDash = char === "-";

        return (
          <div
            key={char + index + "--"}
            className={`flex text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
        justify-${isLetter ? "center" : "flex-end"}
        w-${isLetter ? "8" : "2"}
        border-b-${isLetter ? "4" : "0"}
        items-${
          isLetter || isDash ? "center" : isQuotationMark ? "start" : "end"
        }
        `}
          >
            <p>{!isLetter ? char : isUsed ? char : ""}</p>
          </div>
        );
      })}
    </div>
  );
};

export default Solution;
