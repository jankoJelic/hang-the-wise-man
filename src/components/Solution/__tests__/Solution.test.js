import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Solution from "components/Solution";

const mockSolution = "This is a wise sentence.";

const renderSolution = () =>
  render(
    <Solution solution={mockSolution} usedLetters={["A", "S", "D", "F"]} />
  );

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Solution solution={mockSolution} usedLetters={["A", "S", "D", "F"]} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("words", () => {
  it("groups solution into correct number of words", () => {
    jest.clearAllMocks();
    const wrapper = renderSolution();
    expect(wrapper.getAllByRole("word")).toHaveLength(5);
  });
});

describe("letters", () => {
  it("contains as many letters as in the solution prop, without spaces", () => {
    jest.clearAllMocks();
    const wrapper = renderSolution();

    expect(wrapper.getAllByRole("letter")).toHaveLength(
      mockSolution.replaceAll(" ", "").length
    );
  });
});
