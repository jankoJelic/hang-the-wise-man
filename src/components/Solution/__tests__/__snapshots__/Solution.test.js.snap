// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="w-full flex flex-wrap items-center px-8"
>
  <span
    className="flex flex-row mr-8"
    role="word"
  >
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      >
        s
      </p>
    </span>
  </span>
  <span
    className="flex flex-row mr-8"
    role="word"
  >
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      >
        s
      </p>
    </span>
  </span>
  <span
    className="flex flex-row mr-8"
    role="word"
  >
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      >
        a
      </p>
    </span>
  </span>
  <span
    className="flex flex-row mr-8"
    role="word"
  >
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      >
        s
      </p>
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
  </span>
  <span
    className="flex flex-row mr-8"
    role="word"
  >
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      >
        s
      </p>
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-center
                  w-8
                  border-b-4
                  items-center"
      role="letter"
    >
      <p
        className=""
      />
    </span>
    <span
      className="flex flex-col relative text-mainText font-semibold border-gray-600 mx-1 my-1 h-4 py-4
                  justify-flex-end
                  w-2
                  border-b-0
                  items-end"
      role="letter"
    >
      <p
        className=""
      >
        .
      </p>
    </span>
  </span>
</div>
`;
